flaky examples/relay tests

```
 === RUN   TestMain
      logharness.go:101: saw: Okay, no connection from h1 to h3: no addresses
      logharness.go:106: did not see expected prefix "Meow! It worked!"
  --- FAIL: TestMain (4.54s)

--- a/examples/relay/main_test.go
+++ b/examples/relay/main_test.go
@@ -1,12 +1,16 @@
 package main
 
 import (
+	"os"
 	"testing"
 
 	"github.com/libp2p/go-libp2p/examples/testutils"
 )
 
 func TestMain(t *testing.T) {
+	if os.Getenv("CI") != "" {
+		t.Skip("This test is flaky on CI, see https://github.com/libp2p/go-libp2p/issues/1158.")
+	}
 	var h testutils.LogHarness
 	h.ExpectPrefix("Okay, no connection from h1 to h3")
 	h.ExpectPrefix("Meow! It worked!")

